### PR TITLE
Omar/futurenova

### PR DESCRIPTION
**Pull request by Omarrasheed**
#224

### CODE REVIEW

File: TCAT/AppDelegate.swift [hunk 1]
```diff
@@ -100,6 +105,23 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
     }
+    
+    func setupEndpointConfig() {
+        
+        //
+        // Schemes
+        //
+        
+        // Release - Uses main production server for Network requests.
+        // Debug - Uses development server for Network requests, but will profile and archive to production.
```
Comment (mattbarker016) [inline TCAT/AppDelegate.swift]: The Debug comment actually isn't true, can remove second half

File: Podfile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # Uncomment this line to define a global platform for your project
-platform :ios, '10.0'
+platform :ios, '11.0'
```
Comment (mattbarker016) [inline Podfile]: Is there a reason for this? I thought still target iOS 10 minimum for the whole platform.

Comment (Omarrasheed) [reply Podfile]: FutureNova minimum is 11.0 :(

File: TCAT/Network/Models.swift [hunk 3]
```diff
@@ -93,76 +53,13 @@ class RouteSectionsObject: Codable {
...
-    }
-
-    required init(from decoder: Decoder) throws {
-        let container = try decoder.container(keyedBy: CodingKeys.self)
-        success = try container.decode(Bool.self, forKey: .success)
-        data = try container.decode([Place].self, forKey: .data)
-        parseAllStops()
-    }
-
-    // TO BE MOVED TO BACKEND
```
Comment (mattbarker016) [inline TCAT/Network/Models.swift:117]: This is confirmed moved to backend?

File: TCAT/Model/ServiceAlert.swift [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+//  ServiceAlert.swift
+//  TCAT
+//
+//  Created by Omar Rasheed on 4/19/19.
+//  Copyright © 2019 cuappdev. All rights reserved.
+//
+
+import UIKit
+
+struct ServiceAlert: Codable {
```
Comment (mattbarker016) [inline TCAT/Model/ServiceAlert.swift:11]: Do we have this init for any reason? Can this be simplified?

File: TCAT/Network/Models.swift (hunk 3 above)
Comment (mattbarker016) [reply TCAT/Network/Models.swift:117]: Never mind, I see it was just moved. Should get in backend!

File: TCAT/Controllers/InformationViewController.swift [hunk 5]
```diff
@@ -81,7 +81,7 @@ class InformationViewController: UIViewController, UITableViewDataSource, UITabl
         tcatImage.isUserInteractionEnabled = true
         tcatImage.snp.makeConstraints { (make) in
             make.centerX.equalToSuperview()
-            make.top.equalTo(topLayoutGuide.snp.bottom).offset(44)
+            make.top.equalTo(view.safeAreaLayoutGuide).offset(44)
```
Comment (mattbarker016) [inline TCAT/Controllers/InformationViewController.swift:84]: nice

File: TCAT/Controllers/HomeViewController.swift [hunk 6]
```diff
@@ -46,7 +47,8 @@ class HomeViewController: UIViewController {
     var loadingIndicator: LoadingIndicator?
     var isLoading: Bool { return loadingIndicator != nil }
 
-    let reachability = Reachability(hostname: Network.ipAddress)
+    let reachability = Reachability(hostname: Endpoint.config.host ?? "")
```
Comment (mattbarker016) [inline TCAT/Controllers/HomeViewController.swift:50]: Does FutureNova define this as optional? No way around this?

**Review by mattbarker016 — COMMENTED** [review]
Left some general comments!

File: TCAT/Controllers/HomeViewController.swift (hunk 6 above)
Comment (Omarrasheed) [reply TCAT/Controllers/HomeViewController.swift:50]: yea its an optional from futurenova

File: TCAT/Model/ServiceAlert.swift (hunk 4 above)
Comment (Omarrasheed) [reply TCAT/Model/ServiceAlert.swift:11]: I was using it for dummy data back when i was testing it overseas but I can take it out

File: TCAT/Network/Models.swift (hunk 3 above)
Comment (Omarrasheed) [reply TCAT/Network/Models.swift:117]: Talked to kevin about this. He said he'll do it soon

Comment (mattbarker016) [reply TCAT/Network/Models.swift:117]: @kevinchan159 Do you mind making an issue? Omar or I can clarify anything

**Comment by mattbarker016** [comment]
Awesome. The last thing to check is testing all the endpoints. The hardest things to check are all the cases (success, failure) and endpoints that might have no data or change significantly in their output (live tracking, service alerts), and we want to make sure for every case things are still working accordingly. 

How did you go about testing each endpoint? It might not be entirely reasonable or feasible to absolutely test everything, but how much did you do already?

**Review by doctor3w — APPROVED** [review]
Looks awesome!

**Comment by mattbarker016** [comment]
Merged into #231.

**Pull request closed without merging**